ext/js/accessibility/google-docs-xray.js

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2023  Yomitan Authors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+/** Entry point. */
+function main() {
+    /** @type {Window} */
+    // @ts-expect-error - Firefox Xray vision
+    const window2 = window.wrappedJSObject;
+    if (!(typeof window2 === 'object' && window2 !== null)) { return; }
+    // The extension ID below is on an allow-list that is used on the Google Docs webpage.
+    // @ts-expect-error - Adding a property to the global object
+    // eslint-disable-next-line no-underscore-dangle
+    window2._docs_annotate_canvas_by_ext = 'ogmnaimimemjmbakcfefmnahgdfhfami';
```

Comment:
Isn't this the old yomichan ID? Plus, maybe we should make this dynamic somehow (https://developer.chrome.com/docs/extensions/reference/api/runtime#property-id ?), such that it works regardless of whether you're on stable/dev/unpacked

Reply:
Yomichan's ID is in a whitelist that is hardcoded into the Google's source code, I don't think it works with any random value.

Reply:
Oh... does that mean we can't even fix this unless we ask Google to change their whitelist to point to [`likgccmbimhjbgkjambclfkhldnlhbnn`](https://chromewebstore.google.com/detail/yomitan/likgccmbimhjbgkjambclfkhldnlhbnn)? Or does it not actually matter? It seems I'm a bit confused about how this functionality works...

Reply:
You can search for it in here for example: (the number in the URL probably changes over time)
https://docs.google.com/static/document/client/js/1799341870-client_js_prod_kix_core.js

Reply:
You can use the ID from another extension without issue.

Reply:
The only thing that the google docs script does is check to see if the value of `_docs_annotate_canvas_by_ext` is in their whitelist. It has no way of actually validating that the extension is installed, or that it is the one that set the value. The worst that could happen is that Google denies the extension store submission, but that is very unlikely, especially as this extension is a derivative of the original and there is currently only a testing version publicly available.